### PR TITLE
Refactor/asset stock market index

### PR DESCRIPTION
**Pull request by fmilthaler**
Refactoring the code:
Instead of having almost identical classes for Stock and Market, we now introduce a parent class Asset for both.

### CODE REVIEW

**Comment by fmilthaler** [comment]
@PietropaoloFrisoni Let me know what you think about the new structure. Any feedback is welcome. This is not urgent, so look at it whenever you have time.

File: finquant/stock.py [hunk 1]
```diff
@@ -116,18 +83,12 @@ def properties(self):
         """
         # nicely printing out information and quantities of the stock
         string = "-" * 50
-        string += f"\nStock: {self.name}"
+        string += f"\n{self.asset_type}: {self.name}"
         string += f"\nExpected Return: {self.expected_return:0.3f}"
         string += f"\nVolatility: {self.volatility:0.3f}"
         string += f"\nSkewness: {self.skew:0.5f}"
```
Comment (PietropaoloFrisoni) [inline finquant/stock.py:89]:
```suggestion
        if self.beta is not None:
            string += f"\nStock Beta: {self.beta:0.3f}"
```

File: finquant/stock.py
Comment (PietropaoloFrisoni) [inline finquant/stock.py:1]: @fmilthaler Great improvement! 

What do you think about adding a couple of lines in `properties(self)` function:

`        if self.beta is not None: `
`            string += f"\nStock Beta: {self.beta:0.3f}"` 
?

I should have done it when I submitted my PR introducing the beta parameter, but I forgot : ). I just tried this in the current updated Master Branch, and it seems to work just fine. Please let me know.

**Review by PietropaoloFrisoni — COMMENTED** [review]
Great work! I just have a simple proposal, which I left as a comment in the `stock.py` file.

Best

**Review by PietropaoloFrisoni — APPROVED** [review]

**Pull request merged**